screens/Predict.js

```diff
@@ -0,0 +1,109 @@
+import React, { useState } from 'react'
+import { Dimensions, Modal, SafeAreaView, StyleSheet, Text, View, Image } from 'react-native'
+import Button from '../components/Button'
+import COLOS from '../constants/colors'
+import SIZES from '../constants/fontsize'
+import * as ImagePicker from 'expo-image-picker'
+const height = Dimensions.get('window').height / 3;
+
+const Predict = () => {
+    const [openModal, setOpenModal] = useState(false);
+    const [image, setImage] = useState(null);
+    const uploadImage = async (mode) => {
+        try {
+            let result = {};
+            let permissionsGranted = false;
+
+            if (mode === "gallery") {
+                const { status } = await ImagePicker.requestMediaLibraryPermissionsAsync();
+                permissionsGranted = status === 'granted';
+
+                if (!permissionsGranted) {
+                    alert("Vui lòng bật quyền truy cập trong cài đặt");
+                }
+            } else {
+                const { status } = await ImagePicker.requestCameraPermissionsAsync();
+                permissionsGranted = status === 'granted';
+
+                if (!permissionsGranted) {
+                    alert("Vui lòng bật quyền truy cập trong cài đặt");

```

Comment:
There are duplicate lines of code. Let's reduce `permissionsGranted` and `if` conditions lines